File: Front-End/src/services/order.service.ts
import axiosClient from '@/configurations/axios.config';
import type { OrderCreationRequest, StaffOrderCreationRequest, OrderListResponse, OrderApiResponse, OrderSearchParams } from '@/types/order.type';

export const orderService = {
  createOrder: async (request: OrderCreationRequest) => {
    const response = await axiosClient.post("/orders", request);
    return response.data;
  },

  getMyOrders: async (page: number = 1, size: number = 7, statuses?: string[], startDate?: string, endDate?: string) => {
    const params = new URLSearchParams();
    params.append('page', page.toString());
    params.append('size', size.toString());
    if (statuses && statuses.length > 0) {
      statuses.forEach(status => params.append('status', status));
    }
    if (startDate) params.append('startDate', startDate);
    if (endDate) params.append('endDate', endDate);

    const response = await axiosClient.get<OrderListResponse>(`/orders/my-orders?${params.toString()}`);
    return response.data;
  },

  getAllOrdersForAdmin: async (params: OrderSearchParams) => {
    const queryParams = new URLSearchParams();

    if (params.customerName) queryParams.append('customerName', params.customerName);
    if (params.orderDate) queryParams.append('orderDate', params.orderDate);
    if (params.customerPhone) queryParams.append('customerPhone', params.customerPhone);
    if (params.status) queryParams.append('status', params.status);
    if (params.isPickup !== undefined) queryParams.append('isPickup', params.isPickup.toString());
    queryParams.append('page', (params.page || 1).toString());
    queryParams.append('size', (params.size || 10).toString());

    const response = await axiosClient.get<OrderListResponse>(`/orders/admin?${queryParams.toString()}`);
    return response.data;
  },

  getOrderDetailById: async (id: number) => {
    const response = await axiosClient.get<OrderApiResponse>(`/orders/${id}`);
    return response.data;
  },

  confirmOrder: async (id: number) => {
    const response = await axiosClient.put<OrderApiResponse>(`/orders/${id}/confirm`);
    return response.data;
  },

  cancelOrder: async (id: number) => {
    const response = await axiosClient.put<OrderApiResponse>(`/orders/${id}/cancel`);
    return response.data;
  },

  processOrder: async (id: number) => {
    const response = await axiosClient.put<OrderApiResponse>(`/orders/${id}/process`);
    return response.data;
  },

  completeOrder: async (id: number) => {
    const response = await axiosClient.put<OrderApiResponse>(`/orders/${id}/complete`);
    return response.data;
  },

  getOrdersNeedShipper: async (page: number = 1, size: number = 10) => {
    const response = await axiosClient.get<OrderListResponse>(`/orders/need-shipper`, {
      params: { page, size }
    });
    return response.data;
  },

  staffCreateOrder: async (request: StaffOrderCreationRequest) => {
    const response = await axiosClient.post("/orders/staff-create", request);
    return response.data;
  }
};

File: Back-End/src/main/java/iuh/fit/ecommerce/services/OrderService.java
package iuh.fit.ecommerce.services;

import iuh.fit.ecommerce.dtos.request.order.OrderCreationRequest;
import iuh.fit.ecommerce.dtos.request.order.StaffOrderCreationRequest;
import iuh.fit.ecommerce.dtos.response.base.PageResponse;
import iuh.fit.ecommerce.dtos.response.order.OrderResponse;
import iuh.fit.ecommerce.entities.Order;
import iuh.fit.ecommerce.enums.OrderStatus;
import jakarta.servlet.http.HttpServletRequest;

import java.util.List;

import java.time.LocalDate;

public interface OrderService {
    Object customerCreateOrder(OrderCreationRequest orderCreationRequest, HttpServletRequest request);
    Object staffCreateOrder(StaffOrderCreationRequest request, HttpServletRequest httpRequest);
    PageResponse<OrderResponse> getMyOrders(int page, int size, List<String> status, String startDate, String endDate);

    Order findById(Long id);

    PageResponse<OrderResponse> getAllOrdersForAdmin(
            String customerName, 
            LocalDate orderDate, 
            String customerPhone, 
            OrderStatus status, 
            Boolean isPickup, 
            int page, 
            int size
    );

    OrderResponse getOrderDetailById(Long id);

    OrderResponse confirmOrder(Long orderId);

    OrderResponse cancelOrder(Long orderId);

    OrderResponse processOrder(Long orderId);

    OrderResponse completeOrder(Long orderId);

    PageResponse<OrderResponse> getOrdersNeedShipper(int page, int size);
}
